src/cryptsetup/cryptsetup-generator.c

```diff
@@ -549,7 +586,68 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 free_and_replace(d->keyfile, keyfile);
                 free_and_replace(d->keydev, keydev);
+        } else if (streq(key, "luks.hdrdev")) {
+                size_t n;
+                _cleanup_free_ char *hdrdev = NULL;
+
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                n = strspn(value, ALPHANUMERICAL "-");
+                if (value[n] != '=') {
+                        log_warning("Failed to parse luks.hdrdev= kernel command line switch. UUID is invalid, ignoring.");
+                        return 0;
+                }
+
+                uuid = strndup(value, n);
+                if (!uuid)
+                        return log_oom();
+
+                if (!id128_is_valid(uuid)) {
+                        log_warning("Failed to parse luks.hdrdev= kernel command line switch. UUID is invalid, ignoring.");
+                        return 0;
+                }
```

Comment:
This is very similar to the code in the `luks.key`, please split this out into a common helper function